api/test/e2enew/route/route_with_plugin_http_logger_test.go

```diff
@@ -0,0 +1,239 @@
+						"batch_max_size": 1,
+						"max_retry_count": 1,
+						"retry_delay": 2,
+						"buffer_duration": 2,
+						"inactive_timeout": 2,
+						"name": "http logger",
+						"timeout": 3,
+						"concat_method": "json"
+					}
+				},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": {
+							"` + base.UpstreamIp + `:1982": 1
+						}
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectBody:   []string{`"code":0`, `"id":"r2"`, `"uri":"/hello"`, `"name":"route2"`, `"name":"http logger"`},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("access route to trigger log", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        base.SleepTime,
+		}),
+	)
+
+	ginkgo.It("verify http logger by checking log for second route", func() {
+		// sleep for process log
+		time.Sleep(1500 * time.Millisecond)
+
+		// verify http logger by checking log
+		//todo: should use a fake upstream for confirming whether we got the log data.
+		logContent := base.ReadAPISIXErrorLog()
+		gomega.Expect(logContent).Should(gomega.ContainSubstring("Batch Processor[http logger] failed to process entries: failed to connect to host[127.0.0.1] port[8888] connection refused"))
+
+		// clean log
+		base.CleanAPISIXErrorLog()
+	})
+
+	// todo: check disable http logger - Done
+	table.DescribeTable("test route with log plugin",
+		func(tc base.HttpTestCase) {
+			base.RunTestCase(tc)
+		},
+		table.Entry("disable route http logger plugin", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r2",
+			Body: `{
+				"name": "route2",
+				"uri": "/hello",
+				"plugins": {},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": {
+							"` + base.UpstreamIp + `:1982": 1
+						}
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectBody:   []string{`"code":0`, `"id":"r2"`, `"uri":"/hello"`, `"name":"route2"`},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("access route to trigger log (though should not be triggered)", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        base.SleepTime,
+		}),
+	)
+
+	ginkgo.It("verify http logger has been successfully disabled", func() {
+		// sleep for process log
+		time.Sleep(1500 * time.Millisecond)
+
+		// verify http logger by checking log
+		logContent := base.ReadAPISIXErrorLog()
+		gomega.Expect(logContent).ShouldNot(gomega.ContainSubstring("Batch Processor[http logger] successfully processed the entries"))
+
+		// clean log
+		base.CleanAPISIXErrorLog()
+	})
+
+	table.DescribeTable("test route with log plugin",
```

Comment:
This description confused me.
Maybe delete the route?

Reply:
we could use `clean test data`

Reply:
Awesome!!